Mathlib/Tactic/Bound/Attribute.lean

```diff
@@ -0,0 +1,112 @@
+/-
+Copyright (c) 2024 Geoffrey Irving. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geoffrey Irving
+-/
+import Mathlib.Init.ZeroOne
+import Mathlib.Tactic.Bound.Init
+import Qq
+
+/-!
+# The `bound` attribute
+
+Any lemma tagged with `@[bound]` is registered as an apply rule for the `bound` tactic, by
+converting it to either `norm apply` or `safe apply <priority>`.  The classification is based
+on the number and types of the lemma's hypotheses.
+-/
+
+open Lean (MetaM)
+open Qq
+
+namespace Bound
+
+initialize Lean.registerTraceClass `bound.attribute
+
+variable {u : Lean.Level} {α : Q(Type u)}
+
+/-- Check if an expression is zero -/
+def isZero (e : Q($α)) : MetaM Bool :=
+  match e with
+  | ~q(@OfNat.ofNat.{u} _ 0 $i) => return true
```

Comment:
```suggestion
  | ~q(@OfNat.ofNat.{u} _ (nat_lit 0) $i) => return true
```

Reply:
I removed this line instead, let me know if that's wrong.

Reply:
Fixed the way you suggested, and added tests to verify. Thank you!